content/engineering-practice-layering.md

```diff
@@ -0,0 +1,93 @@
+can also be tested and verified independantly when we employ additional architectual principles such as Dependency Inversion and Encapsulation.
+
+Like the construction 
+example above, this layering represents technology and purpose boundaries, that, when respected, create a modular and consistent design pattern enabling components to be reused and 
+more easily understood.  Indeed, this understanding leads to abstraction of fundamentable behaviours that can organized the code into hierarchies and promote non-duplication of function.
+
+When we layer, we are more prepared for evolving the software as independent peices in an Agile workflow and able to embrace CI/CD and microservice architecture more readily.
+
+---
+
+# Separation of Conerns Principle
+
+Layering is the result of following the *Separation of Concerns* design principle, which states that software should be separated based on the kind
+of work it performs.  When viewed from this lens, we can clearly see what should be in the User Interface collection of code - the code that deals
+with artifacts used to collect or format data for display, and what code performs and enforces business rules and procedures, versus the code used to store and
+retrieve data from a database or file system. 
+
+---
+
+# What are the Layers in Software Architecture
+ 
+Architectually, software applications should be logically constructed following the separation of concerns principle which identifies the application layers.  By separating core business 
+rules and validation logic from user-interface and back-end (data service) infrastructure.  
+
+- **User Interface** - The layer the end-user interacts with, a page or form or visual element such as tables, buttons, charts and graphs, input fields and layout.
+- **Business Logic** - The layer containing the business rules and logical processes, work-flows, and validation rules.
+- **Data Access** - The layer is concerned with the storage and retrieval of data and other artifacts used by the application to perform its funtion.
+
+Each of these layers is focused on a separate logical area of concern, uses different tools and has a different pupose.  Each of
+these logical layers should be created in separate physical projects within the application solution, which helps enforce the separation of concerns principle, improves modularity and testability.  
+
+Ideally, The Business Rules and Logic should not depend on any other projects in the application.
+
+---
+
+# Different Layering Scenarios
+
+As mentioned above, there are logically 3 layers which can be placed into 3 or more physical Tiers enabled by following this software architecture design pattern.  These layers can then be 
+organized into four distict scenarios, shown here.
+
+![Architecture Scenarios](/Images/n-tier-architecture.png)
```

Comment:
@MerlinDarkstar the image files must use the same naming convention as the slug. In other words, please rename your image file to: engineering-practice-layering-1.png